### PR TITLE
fix setting data in output buffers

### PR DESCRIPTION
**Pull request by vhdirk**
This commit fixes reading out the data with CommandQueue.readBuffer, when using anything else than Uint8Arrays.
To correct my earlier pull request: Tutorial 3 and 4 are not ill-formed, the extension just does not (yet) work for asynchronous operations. When changing the tutorials to use blocking queuing operations, it does work as expected.

### CODE REVIEW

**Comment by vhdirk** [comment]
Currently, the only issue left to fix is the behaviour of both commandqueue.finish and webcl.waitForEvents.
They stem from the same, rather difficult to fix, issue:
OpenCL event callbacks may, according to the spec, by triggered from any thread. js-ctypes on the other hand require functions to be called from the same thread they were created in.
To copy with this (see my previous pull request), the asynccallback worker will not register a callback to the webcl event, but rather wait for the event to complete, and then trigger the callback from js, rather than having them triggered from opencl. 

Consequently, when using commandqueue.finish or webcl.waitForEvents, it will wait for all enqueued operations/events to complete, but it will return before the event callbacks are triggered/able to run. I see no easy way of fixing this.

In short: When using blocking enqueue operations, the extension works as expected.

For Firefox >= 42, extension signing is required. Would it be possible to submit this extension to addons.mozilla.org? I do realize that WebCL traction in browser vendors is lacking, but it'd be nice to have this option anyway.
